### PR TITLE
Refactor: Default form

### PR DESCRIPTION
**Pull request by alaca**
## Description

This PR refactors the logic of getting and setting the default form. The default form id is now property of a campaign. The existing `form_id` column is used as a default form id column - this way we don't have to refactor P2P. Also, Campaign's `forms` and `defaultForm` methods are refactored. The `forms` method now returns both V2 and V3 forms. 

## Affects

Campaign migrations
Campaign model
Campaign repository

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: src/Campaigns/Models/Campaign.php [hunk 1]
```diff
@@ -65,22 +67,10 @@ class Campaign extends Model implements ModelCrud, ModelHasFactory
...
-            ->get();
-
-        if (is_null($defaultForm)) {
-            $defaultForm = $this->legacyForms()
-                ->where('campaign_forms.is_default', true)
-                ->get();
-        }
-
-        return $defaultForm;
+        return give(DonationFormsRepository::class)->getById($this->defaultFormId);
```
Comment (JasonTheAdams) [inline src/Campaigns/Models/Campaign.php:73]: Now we're talkin! 🙌

**Comment by JasonTheAdams** [comment]
This is exactly what I was hoping for. This cleaned things up so much. Always good to see a PR that results in less, faster code. 💪

File: src/Campaigns/Migrations/Tables/CreateCampaignFormsTable.php [hunk 2]
```diff
@@ -50,8 +50,7 @@ public function run(): void
 
         $sql = "CREATE TABLE $table (
             campaign_id INT UNSIGNED NOT NULL,
-            form_id INT UNSIGNED NOT NULL,
-            is_default BOOLEAN NOT NULL DEFAULT 0,
+            form_id INT UNSIGNED NOT NULL
```
Comment (kjohnson) [inline src/Campaigns/Migrations/Tables/CreateCampaignFormsTable.php:53]: This is failing automated testing with a SQL syntax error:

```
WordPress database error You have an error in your SQL syntax;
check the manual that corresponds to your MySQL server version
for the right syntax to use at line 4.

CREATE TEMPORARY TABLE wptests_give_campaign_forms (
    campaign_id INT UNSIGNED NOT NULL,
    form_id INT UNSIGNED NOT NULL
    KEY form_id (form_id),
    KEY campaign_id (campaign_id),
    PRIMARY KEY (campaign_id, form_id)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_520_ci
```

Comment (kjohnson) [reply src/Campaigns/Migrations/Tables/CreateCampaignFormsTable.php:53]: Looks like a missing comma.

```suggestion
            form_id INT UNSIGNED NOT NULL,
```

Comment (alaca) [reply src/Campaigns/Migrations/Tables/CreateCampaignFormsTable.php]: Good catch!

**Review by kjohnson — APPROVED** [review]

File: src/Campaigns/Controllers/CampaignRequestController.php [hunk 3]
```diff
@@ -147,6 +146,21 @@ public function updateCampaign(WP_REST_Request $request)
...
+     */
+    public function mergeCampaigns(WP_REST_Request $request): WP_REST_Response
+    {
+        $destinationCampaign = Campaign::find($request->get_param('id'));
+        $campaignsToMerge = Campaign::query()->whereIn('id', $request->get_param('campaignsToMergeIds'))->getAll();
+
+        $campaignsMerged = $destinationCampaign->merge(...$campaignsToMerge);
+
+        return new WP_REST_Response($campaignsMerged);
+    }
```
Comment (glaubersilva) [inline src/Campaigns/Controllers/CampaignRequestController.php:162]: @alaca Not sure if this method is being used in any place. I think the `MergeCampaigns` class is still being used to merge the campaigns.

Comment (alaca) [reply src/Campaigns/Controllers/CampaignRequestController.php:162]: @glaubersilva not sure why that is still there, I remember removing it.

**Pull request merged**